src/net_processing.cpp

```diff
@@ -2684,15 +2689,16 @@ void PeerManagerImpl::HeadersDirectFetchBlocks(CNode& pfrom, const Peer& peer, c
  * whether that header was new and whether the headers message was full,
  * update the state we keep for the peer.
  */
-void PeerManagerImpl::UpdatePeerStateForReceivedHeaders(CNode& pfrom,
+void PeerManagerImpl::UpdatePeerStateForReceivedHeaders(CNode& pfrom, Peer& peer,
         const CBlockIndex *pindexLast, bool received_new_header, bool may_have_more_headers)
 {
+    if (peer.m_unconnecting_headers_msgs > 0) {
+        LogPrint(BCLog::NET, "peer=%d: resetting m_unconnecting_headers_msgs (%d -> 0)\n", pfrom.GetId(), peer.m_unconnecting_headers_msgs);
+    }
+    peer.m_unconnecting_headers_msgs = 0;
```

Comment:
I know this is a move operation, but seems more logical to put it inside the `if` statement imo
```suggestion
    if (peer.m_unconnecting_headers_msgs > 0) {
        LogPrint(BCLog::NET, "peer=%d: resetting m_unconnecting_headers_msgs (%d -> 0)\n", pfrom.GetId(), peer.m_unconnecting_headers_msgs);
        peer.m_unconnecting_headers_msgs = 0;
    }
```